examples/step-3/step-3.cc

```diff
@@ -387,17 +387,18 @@ void Step3::assemble_system ()
   // the type, types::global_dof_index, used here):
   std::vector<types::global_dof_index> local_dof_indices (dofs_per_cell);
 
-  // Now for the loop over all cells. We have seen before how this works, so
-  // the following code should be familiar including the conventional names
-  // for these variables:
-  DoFHandler<2>::active_cell_iterator cell = dof_handler.begin_active();
-  DoFHandler<2>::active_cell_iterator endc = dof_handler.end();
-  for (; cell!=endc; ++cell)
+  // Now for the loop over all cells. We have seen before how this works for a
+  // triangulation. A DoFHandler has cell iterators that are exactly analogous
+  // to those of a Triangulation, but with extra information about the degrees
+  // of freedom for the finite element you're using. Looping over the active
+  // cells of a degree-of-freedom handler works the same as for a triangulation.
+  for (const auto &cell: dof_handler.active_cell_iterators())
     {
-      // @note As already mentioned in step-1, there is a more convenient way
-      // of writing such loops if your compiler supports the C++11
-      // standard. See @ref CPP11 "the deal.II C++11 page" to see
-      // how this works.
+      // Note that we declare the type of the cell as `const auto &` instead of
+      // `auto` this time around. In step 1, we were modifying the cells of the
+      // triangulation by flagging them with refinement indicators. Here we're
+      // only examining the cells without modifying them, so it's good practice
+      // to declare `cell` as `const` in order to enforce this invariant.
```

Comment:
Move this block of five lines above the loop. We've found that it's useful to keep all text that pertains to one particular piece of code together in one place and since you're already discussing the loop above the `for`, it would be worth keeping this comment there as well.

Reply:
I am not sure if the discussion of const vs non-const iterators over cells is important enough for step-1. Generally I care about adding const where possible, but we are not good in how we deal with constness of iterators/accessors of cells. I never bother to use const cell iterators for example.